### ISSUE TITLE
Don't add a hyphen when converting titles containing single quotes to slugs

### ISSUE THREAD

**Issue by raniesantos** [netlify/netlify-cms#1698]
**Is your feature request related to a problem? Please describe.**

Example:

`title: 'First Rule: Don''t Talk About Fight Club'` produces this markdown file `first-rule-don-t-talk-about-fight-club.md`.

**Describe the solution you'd like**

It should be `first-rule-dont-talk-about-fight-club.md`.

It might be a good idea to strip out all single quotes before passing the `title` to whatever function converts it to a `slug`.

**Describe alternatives you've considered**

An alternative solution might be to provide full control of the slug to the user. But that would be a lot more complicated, it means you would need to add a custom slug widget.

**Additional context**

**Comment by erquhart** [comment]
Agreed, and also we _are_ going to have slug editing: https://github.com/netlify/netlify-cms/issues/445

**Comment by tech4him1** [comment]
This can be simply added as another line to `slugFormatter`, next to the existing rule for periods:

https://github.com/netlify/netlify-cms/blob/f801b192211b1e1953ced969a3093eab9e04e14d/packages/netlify-cms-core/src/backend.js#L72-L73
